pkg/controllers/osc/resources/cloud_provider.go

```diff
@@ -41,3 +48,13 @@ func GetCloudProviderFromMachineDeployment(md *clusterv1alpha1.MachineDeployment
 		Spec: *cloudProvider.CloudProviderSpec,
 	}, nil
 }
+
+func GetCloudConfig(secret *corev1.Secret) (string, error) {
+	cloudConfig := secret.Data[CloudConfigConfigName]
+	decodedCloudConfig := make([]byte, base64.StdEncoding.DecodedLen(len(cloudConfig)))
+	if _, err := base64.StdEncoding.Decode(decodedCloudConfig, cloudConfig); err != nil {
```

Comment:
Why? Is the secret value encoded two times?

Reply:
you are right, gonna remove the decoding